.github/workflows/go-test.yml

```
@@ -0,0 +1,24 @@
+name: go-test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-go@v5
+      with:
+        go-version: '1.22'
+    - name: Run unit tests with coverage
+      run: go test -v -coverprofile=coverage.out ./...
```

Давай добавим детектор гонок при прогоне тестов

```suggestion
      run: go test -race -v -coverprofile=coverage.out ./...
```